tests/MavenTests/testEIC.cpp

```diff
@@ -30,130 +23,221 @@ void TestEIC::cleanup() {
 
 void TestEIC::testgetEIC() {
     unsigned int numberOfScans = 445;
-    mzSample* mzsample = new mzSample();
-    mzsample->loadSample(loadFile);
+    mzSample* mzsample = maventests::samples.smallSample;
     EIC e;
 
     bool status = e.makeEICSlice(mzsample, 180.002,180.004, 0, 2, 1, 0, "");
     QVERIFY(e.intensity.size() == numberOfScans && status);
 }
 
 void TestEIC::testgetEICms2() {
-    mzSample* mzsample = new mzSample();
-    mzSample* mzsample_2 = new mzSample();
+    mzSample* mzsample = maventests::samples.ms2TestSamples[1];
+    mzSample* mzsample_2 = maventests::samples.ms2TestSamples[0];
     
-    mzsample->loadSample(files_ms2.at(1).toLatin1().data());
     EIC* e = NULL;
     e = mzsample->getEIC(195,0,70,0,"",0.5,0.5); //precursorMz,collisionEnergy,productMz,eicType,filterline,amuQ1,amuQ3
     QVERIFY(e->rt.size() == 305);
     QVERIFY(e->scannum[e->scannum.size()-1] == 3351);
     QVERIFY(e->maxIntensity == 20200);
 
-    mzsample_2->loadSample(files_ms2.at(0).toLatin1().data());
     EIC* e2 = NULL;
     e2 = mzsample_2->getEIC(195,0,69,1,"",0.5,0.5);
     QVERIFY(e2->rt.size() == 305);
     QVERIFY(e2->scannum[e2->scannum.size()-1] == 3041);
     QVERIFY(e2->maxIntensity == 10600);
 
-    mzsample_2->loadSample(files_ms2.at(0).toLatin1().data());
     EIC* e3 = NULL;
     e3 = mzsample_2->getEIC(195,0,69,1,"",2,2);
     QVERIFY(e3->rt.size() == 305);
-    QVERIFY(e3->scannum[e3->scannum.size()-1] == 6080);
```

Comment:
The scan number changed to half the previous value? why? looks like it's the same file and settings.

Reply:
The value being checked should have been half all this time. The same mzSample object (namely `mzsample_2`) was being used to load the same sample (having 3040 number of scans) file twice. Not sure why this is not handled in `mzSample` class but loading another sample through the same object simply appended the new scans (size: 3040) from second load into the existing scans (size: 3040) vector. Maybe this is intentional, but I fail to see the use case.

Reply:
ahhh. got it. No, it's not intentional.